pkg/cri/server/image_pull.go

```diff
@@ -365,7 +365,7 @@ func (c *criService) registryHosts(ctx context.Context, auth *runtime.AuthConfig
 			var (
 				transport = newTransport()
 				client    = &http.Client{Transport: transport}
-				config    = c.config.Registry.Configs[u.Host]
+				config    = c.config.Registry.Configs[e]
```

Comment:
I think there is something need to be documented.

```go
// https://github.com/containerd/containerd/blob/master/pkg/cri/config/config.go#L156

// Registry is registry settings configured
type Registry struct {
        // Configs are configs for each registry.
	// The key is the domain name or IP of the registry.
	// This option will be fully deprecated for ConfigPath in the future.
	Configs map[string]RegistryConfig `toml:"configs" json:"configs"`
}
```

Check more issues about auth and find that user add the schema in the key for `configs`. 
I think we should document in the config.

```toml
  [plugins."io.containerd.grpc.v1.cri".registry]
        [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:5000"]
          endpoint = ["http://localhost:5000"]
       [plugins."io.containerd.grpc.v1.cri".registry.configs]
          # happy
          [plugins."io.containerd.grpc.v1.cri".registry.configs."localhost:5000".auth]
            username = "testuser"
            password = "testpassword"
            auth = ""
            identitytoken = ""

         # sad
         [plugins."io.containerd.grpc.v1.cri".registry.configs."http://localhost:5000".auth]
            username = "testuser"
            password = "testpassword"
            auth = ""
            identitytoken = ""
```

@pacoxu the change from `u.Host` to `e` might break existing running users. And it also confuses user.

cc @mikebrow @dmcgowan

Reply:
Yes.

As you said, we may document it that mirrors endpoint has to be configured if the endpoint is with a specified port.
```
        [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:5000"]
          endpoint = ["http://localhost:5000"]
```

Reply:
> @pacoxu the change from u.Host to e might break existing running users. And it also confuses user.

For existing running users, 

- if user is using a domain without port, it will still work.
- if user is using IP, it will still work.
- it will break only if they use `no port` in endpoint config but add the default port to the name of the mirror.

```
       [plugins."io.containerd.grpc.v1.cri".registry.mirrors."abc.com:80"]
          endpoint = ["http://abc.com"]
```

Reply:
To handle those backward compability, I prefer adding something like https://github.com/containerd/containerd/compare/master...pacoxu:fix/config?expand=1

Reply:
I think it matter with `scheme` rather than `port`

Reply:
> To handle those backward compability, I prefer adding something like https://github.com/containerd/containerd/compare/master...pacoxu:fix/config?expand=1

it looks good :)

Reply:
I would like to request @mikebrow and @dmcgowan to check this part :)

> My original thought is to add the doc :p